File: Front/RHStoreWS/RHStoreUsuariosBO/TrabajadorBO.cs
using RHStoreBaseBO;
using RHStoreBaseBO.ServiciosWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStoreUsuariosBO
{
	public class TrabajadorBO : BaseBO
	{
        public int insertar(string dni, string nombres, string apellidos, string correo, string contrasenha, string puesto, double sueldo, DateTime fechaIngreso, localTime horarioInicio, localTime horarioFin)
        {
            return TrabajadorWS.insertarTrabajador(dni, nombres, apellidos, correo, contrasenha, puesto, sueldo, fechaIngreso, horarioInicio, horarioFin);
        }

        public int modificar(int idUsuario, string dni, string nombres, string apellidos, string correo, string contrasenha, string puesto, double sueldo, DateTime fechaIngreso, localTime horarioInicio, localTime horarioFin)
        {
            return TrabajadorWS.modificarTrabajador(idUsuario, dni, nombres, apellidos, correo, contrasenha, puesto, sueldo, fechaIngreso, horarioInicio, horarioFin);
        }

        public int eliminar(int idTrabajador)
        {
            return TrabajadorWS.eliminarTrabajador(idTrabajador);
        }

        public BindingList<trabajador> listarTodos()
        {
            return new BindingList<trabajador>(TrabajadorWS.listarTrabajadoresTodos());
        }

        public trabajador obtenerPorId(int idTrabajador)
        {
            return TrabajadorWS.obtenerTrabajadorPorId(idTrabajador);
        }
    }
}

File: Back/SoftRhUsuariosBO/src/pe/edu/pucp/softrh/usuarios/bo/TrabajadorBO.java
package pe.edu.pucp.softrh.usuarios.bo;

import java.time.LocalTime;
import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.usuarios.dao.TrabajadorDAO;
import pe.edu.pucp.softrh.usuarios.daoimp.TrabajadorDAOImp;
import pe.edu.pucp.softrh.usuarios.model.Trabajador;
import pe.edu.pucp.softrh.usuarios.model.Usuario;

public class TrabajadorBO{
	private TrabajadorDAO trabajadorDAO;

	public TrabajadorBO() {
		this.trabajadorDAO = new TrabajadorDAOImp();
	}

	public Integer insertar(String dni, String nombres, String apellidos, String correo, String contrasenha, String puesto, Double sueldo, Date fechaIngreso, String horarioInicio, String horarioFin) {
		Usuario trabajador = new Trabajador(dni, nombres, apellidos, correo, contrasenha, puesto, sueldo, fechaIngreso, horarioInicio, horarioFin);
		return trabajadorDAO.insertar((Trabajador)trabajador);
	}

	public Integer modificar(Integer idUsuario, String dni, String nombres, String apellidos, String correo, String contrasenha, String puesto, Double sueldo, Date fechaIngreso, String horarioInicio, String horarioFin) {
		Usuario trabajador = new Trabajador(dni, nombres, apellidos, correo, contrasenha, puesto, sueldo, fechaIngreso, horarioInicio, horarioFin);
		trabajador.setIdUsuario(idUsuario);
		return trabajadorDAO.modificar((Trabajador)trabajador);
	}

	public Integer eliminar(Integer idtrabajador) {
		return trabajadorDAO.eliminar(idtrabajador);
	}

	public ArrayList<Trabajador> listarTodos() {
		return trabajadorDAO.listarTodos();
	}

	public Trabajador obtenerPorId(Integer idtrabajador) {
		return trabajadorDAO.obtenerPorId(idtrabajador);
	}

	public ArrayList<Trabajador> listarPorDniNombre(String cadena) {
		return trabajadorDAO.listarPorDniNombre(cadena);
	}
}
